seq/src/main.rs

```diff
@@ -0,0 +1,185 @@
+use std::process;
+
+mod cli;
+
+fn main() {
+    let matches = cli::create_app().get_matches();
+
+    if let Some(values) = matches.values_of("FIRST INCREMENT LAST") {
+        let args: Vec<&str> = values.collect();
+        if args.len() > 3 {
+            eprintln!("seq: extra operand '{}'\n Try 'seq --help' for more information.", args[3]);
+            process::exit(1);
+        }
+        let seperator = matches.value_of("SEPERATOR").map(String::from).unwrap();
+        let decimals = max_decimal_digits(&args);
+        let padding = if matches.is_present("WIDTH") { Some(max_digits(&args)) } else { None };
+        let (first, inc, last) = find_operands(&args);
+        let valid_range = (first < last && inc > 0.0) || (first > last && inc < 0.0);
+        if valid_range {
+            let seq = Seq::new(first, inc, last, decimals, seperator, padding);
+            for val in seq.into_iter() {
+                print!("{}", val);
+            }
+            println!("");
+        }
+    } else {
+        eprintln!("seq: missing operand\n Try 'seq --help' for more information.");
+        process::exit(1);
+    }
+}
+
+fn find_operands(args: &[&str]) -> (f64, f64, f64) {
+    match args.len() {
+        1 => (1.0, 1.0, args[0].parse::<f64>().unwrap()),
+        2 => (args[0].parse().unwrap(), 1.0, args[1].parse().unwrap()),
+        _ => (args[0].parse().unwrap(), args[1].parse().unwrap(), args[2].parse().unwrap()),
+    }
```

Comment:
This unwraps aren't ok here, a user may pass something that cannot be parsed into `f64`. The parse case where it is not a valid number should be handled.

Reply:
updated to exit with a useful error message